### PR TITLE
Added parameter to disallow calendar subscription via link

### PR DESCRIPTION
**Pull request by pboguslawski**
Added calendar application config parameter `allow_subscribe_link` to disallow calendar subscription via link (i.e. for internal Nextcloud setups). Use

```
occ config:app:set calendar allow_subscribe_link --value 'yes'
```

to allow (default if not set) and

```
occ config:app:set calendar allow_subscribe_link --value 'no'
```

to disallow calendar subscription via link.

Author-Change-Id: IB#1126265
Signed-off-by: Pawel Boguslawski <pawel.boguslawski@ib.pl>

### CODE REVIEW

File: apps/dav/lib/CalDAV/CalDavBackend.php [hunk 1]
```diff
@@ -2424,6 +2425,10 @@ public function getSubscriptionsForUser($principalUri) {
 	 * @return mixed
 	 */
 	public function createSubscription($principalUri, $uri, array $properties) {
+		// Subscription via link must be enabled.
+		if ($this->config->getAppValue('calendar', 'allow_subscribe_link', 'yes') === 'no') {
```
Comment (tcitworld) [inline apps/dav/lib/CalDAV/CalDavBackend.php:2429]: This should be directly disabled through not registering the plugin here: 
https://github.com/nextcloud/server/blob/754cef369b694c5a58f006ceafe7e9a943c7770f/apps/dav/lib/Server.php#L180

**Review by tcitworld — CHANGES_REQUESTED** [review]

File: apps/dav/lib/Server.php [hunk 2]
```diff
@@ -177,7 +178,9 @@ public function __construct(IRequest $request, string $baseUri) {
 
 			$this->server->addPlugin(\OC::$server->get(\OCA\DAV\CalDAV\Trashbin\Plugin::class));
 			$this->server->addPlugin(new \OCA\DAV\CalDAV\WebcalCaching\Plugin($request));
-			$this->server->addPlugin(new \Sabre\CalDAV\Subscriptions\Plugin());
+			if (\OC::$server->getConfig()->getAppValue('calendar', 'allow_subscribe_link', 'yes') === 'yes') {
```
Comment (tcitworld) [inline apps/dav/lib/Server.php:181]:
```suggestion
			if (\OC::$server->getConfig()->getAppValue('dav', 'allow_subscribe_link', 'yes') === 'yes') {
```
Makes more sense in the `dav` app. Needs https://github.com/nextcloud/calendar/pull/4573 to be adjusted too.

Comment (tcitworld) [reply apps/dav/lib/Server.php:181]: The calendar app is only a CalDAV client, settings in this app should only concern UI stuff. You can use subscriptions without the calendar app (for instance with DAVx5).

To make things clear from other dav stuff I suggest you rename `allow_subscribe_link` in `allow_calendar_subscriptions`.

Comment (pboguslawski) [reply apps/dav/lib/Server.php]: Thanks for clarification. Param name [changed](https://github.com/nextcloud/server/pull/34387/commits/6128163b8a56a7c3f13d7a4f56beb74d6d7622eb) to `dav.allow_calendar_link_subscriptions` to distinguish it from other subscription methods.

**Review by tcitworld — APPROVED** [review]
Please squash your commits

**Comment by pboguslawski** [comment]
> Please squash your commits

Don't see such option in github UI nor have spare time for such cleanup outside now. Please consider [squashing on merge](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-commits) if you still see it valuable or just accept PR as is with all history included.

**Comment by ChristophWurst** [comment]
We've tried to squash the PR for you but we lack the permission to push to your fork.

**Comment by pboguslawski** [comment]
Deleted branch by accident which closed this PR - sorry. Please find squashed PR on
https://github.com/nextcloud/server/pull/34920

**Pull request closed without merging**